hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/keygen/TestComplexKeyGenerator.java

```diff
@@ -178,15 +180,16 @@ void testMultipleValueKeyGenerator(boolean setEncodeSingleKeyFieldNameConfig,
 
   @ParameterizedTest
   @CsvSource(value = {"false,true", "true,false", "true,true"})
-  void testMultipleValueKeyGeneratorNonPartitioned(boolean setEncodeSingleKeyFieldNameConfig,
-                                                   boolean encodeSingleKeyFieldName) {
+  void testMultipleValueKeyGeneratorNonPartitioned(boolean setNewEncodingConfig,
+                                                   boolean encodeSingleKeyFieldValueOnly) {
     TypedProperties properties = new TypedProperties();
     properties.setProperty(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key(), "_row_key,timestamp");
     properties.setProperty(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key(), "");
-    if (setEncodeSingleKeyFieldNameConfig) {
+    properties.setProperty(HoodieWriteConfig.WRITE_TABLE_VERSION.key(), "8");
```

Comment:
For these tests let's also add table version 9 and validate that the `COMPLEX_KEYGEN_NEW_ENCODING` config should not take effect.